presto-main/src/main/java/io/prestosql/server/security/jwt/JwtAuthenticatorSupportModule.java

```diff
@@ -29,10 +32,41 @@ protected void setup(Binder binder)
         configBinder(binder).bindConfig(JwtAuthenticatorConfig.class);
         install(installModuleIf(
                 JwtAuthenticatorConfig.class,
-                config -> config.getKeyFile().startsWith("https://") || config.getKeyFile().startsWith("http://"),
-                jwkBinder -> {
-                    httpClientBinder(jwkBinder)
-                            .bindHttpClient("jwk", null);
-                }));
+                JwtAuthenticatorSupportModule::isHttp,
+                new JwkModule()));
+        install(installModuleIf(
+                JwtAuthenticatorConfig.class,
+                config -> !isHttp(config),
+                jwkBinder -> jwkBinder.bind(SigningKeyResolver.class).to(FileSigningKeyResolver.class).in(Scopes.SINGLETON)));
+    }
+
+    private static boolean isHttp(JwtAuthenticatorConfig config)
+    {
+        return config.getKeyFile().startsWith("https://") || config.getKeyFile().startsWith("http://");
+    }
+
+    private static class JwkModule
+            implements Module
+    {
+        @Override
+        public void configure(Binder binder)
+        {
+            binder.bind(SigningKeyResolver.class).to(JwkSigningKeyResolver.class).in(Scopes.SINGLETON);
+            binder.bind(JwkService.class).in(Scopes.SINGLETON);
+            httpClientBinder(binder)
+                    .bindHttpClient("jwk", ForJwk.class);
+        }
+
+        @Override
+        public int hashCode()
```

Comment:
Where are those methods needed?

Reply:
`hashCode` & `equals`

Reply:
Yes they are needed.  This module can get added by both the web ui and the authentication system, so we need to tell Guice they are the same thing, otherwise you get a duplicate binding error.

Reply:
Please add a comment.